cinema/tests/test_movie_and_movie_session.py

```diff
@@ -0,0 +1,246 @@
+            movie=self.movie_vinston,
+            cinema_hall=self.cinema_hall,
+        )
+
+    def test_movie_if_simple_user(self):
+        movie_response = self.client.get(MOVIE_LIST_URL)
+
+        movies = Movie.objects.all()
+
+        movie_serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(movie_response.status_code, status.HTTP_200_OK)
+
+        self.assertEqual(movie_serializer.data, movie_response.data)
+
+    def test_movie_session_if_simple_user(self):
+        movie_sessions_response = self.client.get(MOVIE_SESSION_LIST_URL)
+
+        movie_sessions = (
+            MovieSession.objects.all()
+            .select_related("movie", "cinema_hall")
+            .annotate(
+                tickets_available=(
+                    F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                    - Count("tickets")
+                )
+            )
+        )
+
+        movie_session_serializer = MovieSessionListSerializer(movie_sessions, many=True)
+
+        self.assertEqual(movie_sessions_response.status_code, status.HTTP_200_OK)
+        self.assertEqual(movie_session_serializer.data, movie_sessions_response.data)
+
+    def test_filter_movie_by_title(self):
+        movie_response_apalon = self.client.get(MOVIE_LIST_URL, {"title": "apalon"})
+        movie_response_vinston = self.client.get(MOVIE_LIST_URL, {"title": "vinston"})
+
+        movie_serializer_apalon = MovieListSerializer(self.movie_apalon, many=False)
+        movie_serializer_vinston = MovieListSerializer(self.movie_vinston, many=False)
+
+        self.assertIn(movie_serializer_apalon.data, movie_response_apalon.data)
+        self.assertIn(movie_serializer_vinston.data, movie_response_vinston.data)
+
+        self.assertNotIn(movie_serializer_vinston.data, movie_response_apalon.data)
+        self.assertNotIn(movie_serializer_apalon.data, movie_response_vinston.data)
+
+    def test_filter_movie_by_actors(self):
+        movie_responce_Illia = self.client.get(
+            MOVIE_LIST_URL, {"actors": f"{self.actor_Illia.id}"}
+        )
+        movie_responce_Jonh = self.client.get(
+            MOVIE_LIST_URL, {"actors": f"{self.actor_John.id}"}
+        )
+
+        movie_serializer_Jonh = MovieListSerializer(self.movie_apalon, many=False)
+        movie_serializer_Illia = MovieListSerializer(self.movie_vinston, many=False)
+
+        self.assertEqual(
+            movie_serializer_Illia.data["actors"],
+            movie_responce_Illia.data[0]["actors"],
+        )
+        self.assertEqual(
+            movie_serializer_Jonh.data["actors"], movie_responce_Jonh.data[0]["actors"]
+        )
+
+        self.assertNotEqual(
+            movie_serializer_Jonh.data["actors"], movie_responce_Illia.data[0]["actors"]
+        )
+
+    def test_filter_movie_by_genres(self):
+        movie_response_horror = self.client.get(
+            MOVIE_LIST_URL, {"genres": f"{self.genre_horror.id}"}
```

Comment:
The same applies here. Add a test case for a genre that does not exist in the database.